core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +826,674 @@ private void thereAndBack(byte[] bytes) {
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.EMPTY_ARRAY),
+        is("[]")
+    );
+    try {
+      SqlFunctions
+          .jsonQuery(
+              SqlFunctions.PathContext
+                  .withReturned(SqlFunctions.PathMode.STRICT, "bar"),
+              SqlJsonQueryWrapperBehavior.WITHOUT_ARRAY,
+              SqlJsonQueryEmptyOrErrorBehavior.NULL,
+              SqlJsonQueryEmptyOrErrorBehavior.ERROR);
+      fail("expect exception, but not");
+    } catch (Exception e) {
+      assertThat(
+          e.getMessage(), is("java.lang.RuntimeException: "
+          + "not a json array or a json object: bar"));
+    }
+
+    // wrapper behavior test
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, "bar"),
+                SqlJsonQueryWrapperBehavior.WITH_UNCONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, "bar"),
+                SqlJsonQueryWrapperBehavior.WITH_CONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.singletonList("bar")),
+                SqlJsonQueryWrapperBehavior.WITH_UNCONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[[\"bar\"]]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.singletonList("bar")),
+                SqlJsonQueryWrapperBehavior.WITH_CONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+  }
+
+  @Test public void testJsonize() {
+    assertThat(
+        SqlFunctions.jsonize(new HashMap<>()),
+        is("{}")
+    );
+  }
+
+  @Test public void testDejsonize() {
+    assertThat(
+        SqlFunctions.dejsonize("{}"),
+        is(Collections.emptyMap())
+    );
+    assertThat(
+        SqlFunctions.dejsonize("[]"),
+        is(Collections.emptyList())
+    );
+
+    // expect exception thrown
+    try {
+      SqlFunctions.dejsonize("[}");
+      fail("expect exception, but not");
+    } catch (Exception ignored) {
+      // ignored
+    }
+  }
+
+  @Test public void testJsonObject() {
```

Comment:
Would you please ensure the tests print human-understandable messages in case of failures?